tests/test_idpi/conftest.py

```diff
@@ -10,28 +10,105 @@
 from jinja2 import Environment, FileSystemLoader
 
 
-@pytest.fixture
-def data_dir():
+@pytest.fixture(scope="session")
+def machine():
+    """Machine name."""
+    # adapted from spack-c2sm
+    path = Path("/etc/xthostname")
+    if path.exists():
+        return path.read_text().strip()
+    hostname = subprocess.check_output(["hostname", "-s"]).decode()
+    match hostname.split("-"):
+        case "tsa", *_:
+            return "tsa"
+        case "arolla", *_:
+            return "arolla"
+    return "unknown"
+
+
+@pytest.fixture(scope="session")
+def data_dir(machine, tmp_path_factory):
     """Base data dir."""
-    return Path("/project/s83c/rz+/icon_data_processing_incubator/data/SWISS")
+    if machine == "tsa":
+        return Path("/project/s83c/rz+/icon_data_processing_incubator/data/SWISS")
+    if machine == "balfrin":
+        return tmp_path_factory.mktemp("data")
+    raise RuntimeError("panic")
 
 
-@pytest.fixture
-def fieldextra_executable():
-    """Fieldextra executable."""
-    return "/project/s83c/fieldextra/tsa/bin/fieldextra_gnu_opt_omp"
+@pytest.fixture(scope="session")
+def fieldextra_path(machine):
+    """Fieldextra path."""
+    conf = {
+        "tsa": Path("/project/s83c/fieldextra/tsa"),
+        "balfrin": Path("/users/tsm/proj.aare/fieldextra/v14.3.1/"),
+    }
+    return conf[machine]
 
 
-@pytest.fixture
+@pytest.fixture(scope="session")
 def template_env():
     """Jinja input namelist template environment."""
     test_dir = Path(__file__).parent
     loader = FileSystemLoader(test_dir / "fieldextra_templates")
     return Environment(loader=loader, keep_trailing_newline=True)
 
 
+@pytest.fixture(scope="session")
+def download_data(work_dir, machine):
+    import os
+
+    os.environ["FDB5_DIR"] = (
+        "/scratch/ckanesan/spack-c2sm/spack/opt/spack/"
+        "__spack_path_placeholder__/__spack_path_placeholder__/"
+        "__spack_path_placeholder__/__spack_path_placeholder__/"
+        "__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placehold/"
+        "fdb/5.11.17/gcc/nisooybvm5rkt43sqimbsn75rtmh64ri"
+    )
```

Comment:
Could you not also have a fixture for fdb_dir with balfrin/tsa keys like you have for the fieldextra path?

Reply:
This has been replaced with a view so the machine specific part should be handled by spack.